helpers/calendar/events.py

```diff
@@ -90,3 +90,50 @@ def get_all_room_schedules(self, query, start_date, end_date):
                 all_events.append(current_event)
 
         return all_events, all_dates
+
+    def get_room_index(self, attendees, calendar_id):
+        index_of_room = 0
+        for index in range(len(attendees)):
+            if attendees[index]["email"] == calendar_id:
+                index_of_room = index
+                break
+        return index_of_room
+
+    def get_all_recurring_events(self, query, start_date, end_date):
+        """ get all recurring events"""
+        rooms = query.filter_by(state="active")
+        credentials = Credentials()
+        service = credentials.set_api_credentials()
+        recurring_events = []
+        for room in rooms:
+            events = []
+            try:
+                events = service.events().list(
+                    calendarId=room.calendar_id, timeMax=end_date,
+                    timeMin=start_date, singleEvents=True,
+                    orderBy='startTime').execute()
+            except Exception:
+                continue
+            for event in events['items']:
+                check_recurring_event = event.get("recurringEventId")
+                if not event.get("attendees"):
+                    continue
+                room_index = self.get_room_index(
+                    event["attendees"],
+                    room.calendar_id
+                    )
+                room_declined = event.get("attendees")[room_index]["responseStatus"]  # noqa: E501
```

Comment:
Does room_declined refer to meeting/event declined? Kindly explain what this means

Reply:
room_declined is  a variable that stores the status of the response the status gave. Now that I am saying it I think we can update to be room_response_status. We will implement that shortly.